File: easy/3833_count_dominant_indices/solution.cpp
#include <vector>

class Solution {
public:
  int dominantIndices(std::vector<int> &nums) {
    int postfixSum = nums.back();
    int postfixCount = 1;

    int count = 0;

    for (int i = nums.size() - 2; i >= 0; --i) {
      if (nums[i] * postfixCount > postfixSum)
        ++count;

      postfixSum += nums[i];
      ++postfixCount;
    }

    return count;
  }
};

File: easy/3833_count_dominant_indices/solution.py
from typing import List


class Solution:
    def dominantIndices(self, nums: List[int]) -> int:
        postfix_sums = [0] * len(nums)

        for i in range(len(nums) - 2, -1, -1):
            postfix_sums[i] = postfix_sums[i + 1] + nums[i + 1]

        count = 0

        for i in range(len(nums) - 1):
            if nums[i] > postfix_sums[i] / (len(nums) - i - 1):
                count += 1

        return count
